<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Nociones Básicas de Bioinformática y Genómica  Máster en Bioinformática, Universidad de Valencia" />
  <title>Practical: Gene Ontology</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      /*font-style:italic;*/
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">Practical: Gene Ontology</h1>
<h2 class="author">Nociones Básicas de Bioinformática y Genómica <br> <em><a href="http://www.uv.es/bioinfor/">Máster en Bioinformática, Universidad de Valencia</a></em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2015-05-18</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#overview">Overview</a></li>
<li><a href="#get-gene-annotation-form-ensembl">Get gene annotation form Ensembl</a></li>
<li><a href="#amigo-search">Amigo Search</a></li>
<li><a href="#amigo-tools">Amigo Tools</a></li>
</ul>
</div>
<hr>
<!--
Clean the starting dataset
================================================================================
#    rm -r data/*
#    cp data0/f* data/
#    cd data
-->
<!-- links -------------------------------------------------------------------->
<h1 id="overview">Overview</h1>
<p>In this practical we will explore the structure of the Gene Ontology data base and see how to query gene <em>annotations</em>.</p>
<h1 id="get-gene-annotation-form-ensembl">Get gene annotation form Ensembl</h1>
<p>Go to the <a href="http://www.ensembl.org/" title="Ensembl home">Ensembl</a> <em>biomart</em> web site and download all human genes and transcripts.</p>
<ul>
<li>How many different gene ids are there?</li>
<li>How many different transcripts ids are there?</li>
<li>Is there any transcript associated to more than one gene?</li>
</ul>
<p>Go to the <a href="http://www.ensembl.org/" title="Ensembl home">Ensembl</a> <em>biomart</em> web site and download the GO annotation for all human genes.</p>
<p>Get the following fields</p>
<ul>
<li>Ensembl Gene ID</li>
<li>Ensembl Transcript ID</li>
<li>Chromosome Name</li>
<li>GO Term Accession</li>
<li>GO Term Name</li>
<li>GO Term Evidence Code</li>
<li>GO domain</li>
</ul>
<!-- 
- GO Term Definition
-->
<p><strong>Which information is reported in each of them?</strong></p>
<p>Use <strong>R</strong> to read the information and explore the data. Answer the following questions:</p>
<ul>
<li><p>Which percentage of genes are annotated?</p></li>
<li><p>Which percentage of transcripts are annotated?</p></li>
<li><p>Are there <em>duplicated</em> annotations (same gene or transcript annotated to the same GO term)? Why?</p></li>
<li><p>Which percentage of the gene annotations belong to each ontology? (Exclude duplications caused by the the Evidence codes)</p></li>
<li><p>Which percentage of the go terms belong to each ontology? (Exclude duplications caused by the the Evidence codes)</p></li>
<li><p>Why are there so few <em>cellular component</em> annotations?</p></li>
<li><p>Explore the distribution of the number of annotations of each gene.<br /> HINT: get the count of the number of GO terms annotated to each gene. Make some <em>Box plots</em>.</p></li>
</ul>
<!--
- Compare those distributions across chromosomes.
-->
<h1 id="amigo-search">Amigo Search</h1>
<p>Go to the <a href="http://amigo.geneontology.org/amigo">Amigo Search page</a> and find the GO id for the term: <em>regulation of execution phase of apoptosis</em>.</p>
<!-- GO:1900117 -->
<ul>
<li>Which is the ontology of this term?</li>
</ul>
<p>Find all the human genes associated to this term. (HINT: use the <em>Taxon filter</em>)</p>
<ul>
<li><p>How many are they?</p></li>
<li><p>Why are there several duplicates? <!-- there are several protein synonyms --></p></li>
</ul>
<p>Download the table</p>
<ul>
<li>Why is the URL so long?</li>
</ul>
<p>Get the same term in the data you did download from Ensembl.</p>
<ul>
<li>Can you read the Amigo data into R?</li>
</ul>
<h1 id="amigo-tools">Amigo Tools</h1>
<p>Go to the Amigo Tools web page and find the visualization tool.</p>
<p>Find several GO terms related to apoptosis form your ensembl dataset or form the Amigo web page (HINT: you can use the R command <code>grep</code>) and plot them in the visualization tool.</p>
<p>Try to include some colors in the graph.</p>
<!-- 
http://amigo.geneontology.org/amigo/software_list

http://amigo.geneontology.org/amigo/visualize

http://amigo1.geneontology.org/cgi-bin/amigo/term_enrichment
-->
</body>
</html>
