<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Nociones Básicas de Bioinformática y Genómica  Máster en Bioinformática, Universidad de Valencia" />
  <title>Practical: Gviz</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      /*font-style:italic;*/
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">Practical: Gviz</h1>
<h2 class="author">Nociones Básicas de Bioinformática y Genómica <br> <em><a href="http://www.uv.es/bioinfor/">Máster en Bioinformática, Universidad de Valencia</a></em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2015-05-25</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#data-download">Data Download</a></li>
<li><a href="#data-display">Data Display</a><ul>
<li><a href="#data-reading">Data Reading</a></li>
<li><a href="#data-formatting">Data Formatting</a></li>
<li><a href="#graphical-display">Graphical Display</a></li>
</ul></li>
</ul>
</div>
<hr>
<p>The aim of this practical is to get a representation of all transcripts of the gene <em>SMN1</em> in human. In this case we will extract the data form the Ensembl database using Biomart web page, and we will use the R/Bioconductor package <code>Gviz</code> to make the graphical display of the data.</p>
<h1 id="data-download">Data Download</h1>
<p>Although we are now just interested in <em>SMN1</em>, it takes the same effort to prepare the data only for this gene or for all genes in the human genome. Thus, we will download and prepare data for all human genes.</p>
<p>We will need to download the following information:</p>
<ul>
<li><p>Gene name</p></li>
<li><p>Transcript Id</p></li>
<li><p>Exon Id</p></li>
<li><p>Chromosome</p></li>
<li><p>Exon end position</p></li>
<li><p>Exon end position</p></li>
<li><p>Strand (not necessary at the moment but must be interesting to keep it)</p></li>
</ul>
<p>Remember that in Ensembl all information is ultimately linked to the <em>Ensemble Gene Identifier</em> so we may also need to use this field to combine all information.</p>
<p>Go to the Ensembl web page at <a href="http://www.ensembl.org/" class="uri">http://www.ensembl.org/</a> and follow the BioMart link. The information we are looking for is in the <em>Genes Database</em>. Once you select the database and the species you will be able to choose the <em>Attributes</em> to be downloaded.</p>
<p>The information about exons and their position is in the <strong>Structures</strong> section. As all tables in Ensembl, this table is ultimately linked to <em>Ensembl Gene ID</em>, hence, you will also need to find the information for the conversion between <em>Ensembl Gene ID</em> and <em>Gene Names</em>.</p>
<p>The conversion between Ensembl Gene IDs and all other external databases is in the section <strong>Features</strong> of the <em>Attributes</em> section. From this section you will need to retrieve the <em>HGNC symbol</em> which is the “common” gen name.</p>
<p>Once you have downloaded the <strong>two tables</strong> we are ready to format the data and plot our gene.</p>
<h1 id="data-display">Data Display</h1>
<h2 id="data-reading">Data Reading</h2>
<p>Read the data downloaded form Ensembl using the <code>R</code> function <code>read.table</code>.</p>
<p>This parameters of the function are generally suitable:</p>
<ul>
<li><p>header = TRUE</p></li>
<li><p>sep = “\t”</p></li>
<li><p>quote = “”</p></li>
<li><p>as.is = TRUE</p></li>
</ul>
<h2 id="data-formatting">Data Formatting</h2>
<p>Combine the two data frames using the <code>R</code> function <code>merge</code>.</p>
<p>Keep just the rows of the combined dataset which have information for <em>SMN1</em>. You can “cut” the data frame doing</p>
<p><code>datos[datos$HGNC.symbol == SMN1,]</code></p>
<h2 id="graphical-display">Graphical Display</h2>
<p>Load the <code>R</code> library <code>Gviz</code> and define a track to store the information for the genes using the function, <code>GeneRegionTrack</code>. Find the help for this function in writing <code>?GeneRegionTrack</code> in the <code>R</code> command line if you need it.</p>
<p>Plot the data using the function <code>plotTracks</code>.</p>
<p>Save the plot using the <code>R</code> function <code>png</code></p>
</body>
</html>
