<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Nociones Básicas de Bioinformática y Genómica  Máster en Bioinformática, Universidad de Valencia" />
  <title>Practical: Ensembl</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
      /*font-style:italic;*/
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
      line-height:1.5em;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  /* animated spoiler CSS by Bloggersentral.com */
  .spoilerbutton {
  	display:block;
  	margin:5px 0;
  	padding: 3px 6px;
  	color: #20639b;
  	font-size: small;
  	font-weight: bold;
  }
  .spoiler {overflow:hidden;background: #f5f5f5;}
  .spoiler > div {-webkit-transition: all 0.2s ease;-moz-transition: margin 0.2s ease;-o-transition: all 0.2s ease;transition: margin 0.2s ease;}
  .spoilerbutton[value="Show answer"] + .spoiler > div {margin-top:-100%;}
  .spoilerbutton[value="Hide answer"] + .spoiler {padding:5px;} 
  /**********************************************/
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">Practical: Ensembl</h1>
<h2 class="author">Nociones Básicas de Bioinformática y Genómica <br> <em><a href="http://www.uv.es/bioinfor/">Máster en Bioinformática, Universidad de Valencia</a></em></h2>
<h3 class="date"><a href="http://www.dmontaner.com">David Montaner</a> <br> <em>2015-03-30</em></h3>
</div>
<hr>
<div id="TOC">
<ul>
<li><a href="#exercise-1">Exercise 1</a></li>
<li><a href="#exercise-2">Exercise 2</a></li>
<li><a href="#exercise-3">Exercise 3</a></li>
<li><a href="#exercise-4">Exercise 4</a></li>
</ul>
</div>
<hr>
<h1 id="exercise-1">Exercise 1</h1>
<ol style="list-style-type: decimal">
<li>Find all <em>short variants</em> of the human gene <strong>BRCA2</strong> in BioMart. Hint: find out the Ensembl ID for the gene and set a filter when searching in BioMart.
<ul>
<li>How many are they?</li>
<li>Are they all SNPs?</li>
</ul></li>
<li>Download the <em>somatic short variants</em> associated to gene <strong>BRCA2</strong> and find out if any ID is shared with the <em>short variation</em> database. (Some example R code is provided for this task).</li>
<li>Download the sequence of the human gene <strong>BRCA2</strong>. (Go to Attributes &gt; Sequences)</li>
<li>Download the genomic position (chromosome, start and end) of all exons of the gene. (Go to Attributes &gt; Structures)</li>
</ol>
<!-- 
BRCA2 = ENSG000001396185820 short in BRCA2
1035 somatic in BRCA2
0 structural in BRCA2

5024 structural in human

-->
<h1 id="exercise-2">Exercise 2</h1>
<ol style="list-style-type: decimal">
<li>How many <em>structural variants</em> are there registered Ensembl for in humans?</li>
<li>How many <em>somatic structural variants</em> are there registered Ensembl for in humans? Download them from BioMart file and find out all possible <em>types</em>. Use Linux shell commands: <code>cut</code>, <code>sort</code> and <code>uniq</code>: <code>cut -f 6  mart_export_somatic_structural.txt | sort | uniq</code></li>
</ol>
<!--
cut -f 6 mart_export_structural.txt | sort | uniq

complex_structural_alteration
copy_number_variation
insertion
interchromosomal_breakpoint
intrachromosomal_breakpoint
inversion
mobile_element_insertion
SV Variant Type
tandem_duplication
-->
<h1 id="exercise-3">Exercise 3</h1>
<p>Find out the Ensembl FTP site form which you can download the <strong>C. elegans</strong> reference genome.</p>
<ul>
<li>How many chromosomes does the worm have?</li>
<li>Why are there three <em>mitochondrial</em> files in the FTP site? Indeed all chromosomes have triplicated files. Find out their differences in the README file.</li>
<li>Download all three <em>mitochondrial</em> files and explore their differences. Use <code>wget</code> and <code>colordiff</code>.</li>
<li>What is it in the <em>CHECKSUMS</em> file?</li>
<li>Download the <em>toplevel</em> file and make sure that all chromosomes are represented within it. Use <code>grep</code> to extract sequence names from the file. <code>grep &quot;&gt;&quot; Caenorhabditis_elegans.WBcel235.dna.chromosome.I.fa</code></li>
<li>Find out in the Ensembl FTP site the <a href="http://www.ensembl.org/info/website/upload/gff.html">GTF</a> file annotationg this genome.</li>
</ul>
<!-- 
http://www.ensembl.org/Caenorhabditis_elegans/Info/Index
Download DNA sequence (FASTA)

wget ftp://ftp.ensembl.org/pub/release-79/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna.chromosome.MtDNA.fa.gz
wget ftp://ftp.ensembl.org/pub/release-79/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna_rm.chromosome.MtDNA.fa.gz
wget ftp://ftp.ensembl.org/pub/release-79/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna_sm.chromosome.MtDNA.fa.gz

wget ftp://ftp.ensembl.org/pub/release-79/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna.toplevel.fa.gz


>I dna:chromosome chromosome:WBcel235:I:1:15072434:1 REF
>II dna:chromosome chromosome:WBcel235:II:1:15279421:1 REF
>III dna:chromosome chromosome:WBcel235:III:1:13783801:1 REF
>IV dna:chromosome chromosome:WBcel235:IV:1:17493829:1 REF
>MtDNA dna:chromosome chromosome:WBcel235:MtDNA:1:13794:1 REF
>V dna:chromosome chromosome:WBcel235:V:1:20924180:1 REF
>X dna:chromosome chromosome:WBcel235:X:1:17718942:1 REF

ftp://ftp.ensembl.org/pub/release-79/gtf/caenorhabditis_elegans/

-->
<h1 id="exercise-4">Exercise 4</h1>
<p>Download the expression matrix for the <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE35972">GSE35972</a> dataset of the <a href="http://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus</a> at the NCBI.</p>
<p>Hint: download the <strong>Series Matrix File(s)</strong> form the <a href="ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE35nnn/GSE35972/matrix/">FTP site</a></p>
<p>As you can see in the associated <em>platform</em> <a href="www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL570">GPL570</a>, the <em>GSE35972</em> experiment used an <strong>Affymetrix Human Genome U133 Plus 2.0 Array</strong> to measure gene expression.</p>
<p>Explore the file and find out the <em>probe</em> identifiers.</p>
<p>Find out in BioMart the ID conversion between the Affymetrix <em>probe</em> IDs and the <a href="http://www.genenames.org/">Gene Names</a>. Create a matrix file with the ID conversion. (Some example R code is provided for this task).</p>
<p>Find out the <a href="http://geneontology.org">Gene Ontology Annotation</a> for this data set <strong>using BioMart</strong>.</p>
</body>
</html>
